common/resource.go

```diff
@@ -199,7 +199,11 @@ func DataResource(sc any, read func(context.Context, any, *DatabricksClient) err
 				diags = diag.FromErr(err)
 			}
 			StructToData(ptr.Elem().Interface(), s, d)
-			d.SetId("_")
+			if _, ok := s["id"]; ok {
```

Comment:
please add more extended comments - it's clear to me and you, but possibly not to others :)